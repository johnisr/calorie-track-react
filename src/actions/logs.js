import database from '../firebase/firebase';
import moment from 'moment';

export const defaultLog = {
  date: 0,
  weight: 0,
  unit: 'lb',
  foods: [],
};

export const addLog = (log) => ({
  type: 'ADD_LOG',
  log,
});

export const startAddLog = (data = {}) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    const {
      weight = 0,
      unit = 'lb',
      foods = [],
      date = moment(),
    } = data;
    const log = { weight, unit, foods, date };

    try {
      await database.ref(`users/${uid}/logs/${date}`).set(log);
      dispatch(addLog(log));
    } catch (e) {

    };
  };
};

export const removeLog = (date) => ({
  type: 'REMOVE_LOG',
  date,
});

export const startRemoveLog = (date) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    try {
      await database.ref(`users/${uid}/logs/${date}`).remove();
      dispatch(removeLog(date));
    } catch (e) {

    };
  };
};

export const editLog = (date, updates) => ({
  type: 'EDIT_LOG',
  date,
  updates,
});

export const startEditLog = (date, updates) => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;
    try {
      await database.ref(`users/${uid}/logs/${date}`).update(updates);
      dispatch(editLog(date, updates));
    } catch (e) {
      
    }
  };
};

export const setLogs = (logs) => ({
  type: 'SET_LOGS',
  logs,
});

export const startSetLogs = () => {
  return async (dispatch, getState) => {
    const { uid } = getState().auth;

    try {
      const snapshot = await database.ref(`users/${uid}/logs`).once('value');
      const logs = [];
      snapshot.forEach((log) => {
        logs.push(log.val());
      });
      dispatch(setLogs(logs));
    } catch (e) {

    }
  };
};