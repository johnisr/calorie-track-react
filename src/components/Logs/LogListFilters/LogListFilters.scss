@import '../../../styles/base';

.logsListFilters {
  grid-column: center-start / center-end;
  justify-content: center;
  align-items: center;
  padding: 2rem;

  background-color: $color-grey-dark-2;
  display: flex;

  &__search {
    flex: 0 0 50%;
  }

  &__input {
    font-size: $font-size-quarternary;
    color: $color-grey-dark-1;
    width: 100%;
    height: 3rem;
    background-color: $color-grey-light-3;

    border: none;
  }

  &__options {
    margin-left: 1rem;
    height: 3rem;
    display: flex;

    @media only screen and (max-width: $bp-smaller) {
      font-size: $font-size-body;
      flex: 0 0 15%;
    }
    
  }
  
  &__select {
    color: $color-grey-dark-1; 
    font-size: $font-size-quarternary;
    background-color: $color-grey-light-3;
    border: none;

    @media only screen and (max-width: $bp-smaller) {
      font-size: $font-size-body;
      width: 40%;
    }
  }

  &__pages {
    display: flex;
    height: 3rem;
  }
}