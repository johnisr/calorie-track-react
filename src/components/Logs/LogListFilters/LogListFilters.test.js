import React from 'react';
import { shallow } from 'enzyme';
import moment from 'moment';
import { LogListFilters } from './LogListFilters';
import logs from '../../../tests/fixtures/logs';

let filters, logsLength, setStartDate, setEndDate, sortByNewest, sortByOldest, setMaxLogsShown, setLogsOffset, wrapper;
beforeEach(() => {
  filters = {
    startDate: moment(0),
    endDate: moment(1000),
    sortBy: 'newest',
    maxLogsShown: 7,
    offset: 0,
  };
  logsLength = logs.length;
  setStartDate = jest.fn();
  setEndDate = jest.fn();
  sortByNewest = jest.fn();
  sortByOldest = jest.fn();
  setMaxLogsShown = jest.fn();
  setLogsOffset = jest.fn();
  wrapper = shallow(<LogListFilters
    filters={filters}
    logsLength={logsLength}
    setStartDate={setStartDate}
    setEndDate={setEndDate}
    sortByNewest={sortByNewest}
    sortByOldest={sortByOldest}
    setMaxLogsShown={setMaxLogsShown}
    setLogsOffset={setLogsOffset}
  />);
});

it('should render correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should handle date changes', () => {
  const startDate = moment(0).add(4, 'years');
  const endDate = moment(0).add(8, 'years');
  wrapper.find('withStyles(DateRangePicker)')
    .prop('onDatesChange')({ startDate, endDate });
  expect(setStartDate).toHaveBeenCalledWith(startDate);
  expect(setEndDate).toHaveBeenCalledWith(endDate);
});

it('should handle date focus changes', () => {
  const calendarFocused = 'startDate';
  wrapper.find('withStyles(DateRangePicker)')
    .prop('onFocusChange')(calendarFocused);
  expect(wrapper.state('calendarFocused')).toBe(calendarFocused);
});

it('should change sort by to newest', () => {
  const value = 'newest';
  const e = {target: { value }};
  wrapper.find('select').at(0).simulate('change', e);
  expect(sortByNewest).toHaveBeenCalled();
})

it('should change sort by to oldest', () => {
  const value = 'oldest';
  const e = {target: { value }};
  wrapper.find('select').at(0).simulate('change', e);
  expect(sortByOldest).toHaveBeenCalled();
});

it('should change max logs shown', () => {
  const value = '14';
  const e = {target: { value }};
  wrapper.find('select').at(1).simulate('change', e);
  expect(setMaxLogsShown).toHaveBeenLastCalledWith(14);
  expect(setLogsOffset).toHaveBeenLastCalledWith(0);
});

it('should change off set with prev page click', () => {
  wrapper.find('button').at(0).simulate('click');
  expect(setLogsOffset).toHaveBeenLastCalledWith(filters.offset - 1);
});

it('should change off set with next page click', () => {
  wrapper.find('button').at(2).simulate('click');
  expect(setLogsOffset).toHaveBeenLastCalledWith(filters.offset + 1);
});

// Buttons: [Prev Page] [1] [Next Page]
it('should change off set with next page click', () => {
  wrapper.find('button').at(1).simulate('click');
  expect(setLogsOffset).toHaveBeenLastCalledWith(0);
});

