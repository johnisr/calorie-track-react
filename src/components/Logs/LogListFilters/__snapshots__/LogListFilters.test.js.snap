// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LogListFilters
    filters={
        Object {
            "endDate": "1970-01-01T00:00:01.000Z",
            "maxLogsShown": 7,
            "offset": 0,
            "sortBy": "newest",
            "startDate": "1970-01-01T00:00:00.000Z",
          }
    }
    logsLength={3}
    setEndDate={[Function]}
    setLogsOffset={[Function]}
    setMaxLogsShown={[Function]}
    setStartDate={[Function]}
    sortByNewest={[Function]}
    sortByOldest={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <withStyles(DateRangePicker)
          anchorDirection="left"
          block={false}
          calendarInfoPosition="bottom"
          customArrowIcon={null}
          customCloseIcon={null}
          customInputIcon={null}
          daySize={39}
          disabled={false}
          displayFormat={[Function]}
          enableOutsideDays={false}
          endDate={"1970-01-01T00:00:01.000Z"}
          endDateId="endDate"
          endDatePlaceholderText="End Date"
          firstDayOfWeek={null}
          focusedInput={null}
          hideKeyboardShortcutsPanel={false}
          horizontalMargin={0}
          initialVisibleMonth={null}
          inputIconPosition="before"
          isDayBlocked={[Function]}
          isDayHighlighted={[Function]}
          isOutsideRange={[Function]}
          isRTL={false}
          keepFocusOnInput={false}
          keepOpenOnDateSelect={false}
          minimumNights={1}
          monthFormat="MMMM YYYY"
          navNext={null}
          navPrev={null}
          noBorder={false}
          numberOfMonths={1}
          onClose={[Function]}
          onDatesChange={[Function]}
          onFocusChange={[Function]}
          onNextMonthClick={[Function]}
          onPrevMonthClick={[Function]}
          openDirection="down"
          orientation="horizontal"
          phrases={
                    Object {
                              "calendarLabel": "Calendar",
                              "chooseAvailableEndDate": [Function],
                              "chooseAvailableStartDate": [Function],
                              "clearDates": "Clear Dates",
                              "closeDatePicker": "Close",
                              "dateIsSelected": [Function],
                              "dateIsUnavailable": [Function],
                              "enterKey": "Enter key",
                              "escape": "Escape key",
                              "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
                              "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
                              "homeEnd": "Home and end keys",
                              "jumpToNextMonth": "Move forward to switch to the next month.",
                              "jumpToPrevMonth": "Move backward to switch to the previous month.",
                              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
                      select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
                              "keyboardShortcuts": "Keyboard Shortcuts",
                              "leftArrowRightArrow": "Right and left arrow keys",
                              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
                              "moveFocusByOneMonth": "Switch months.",
                              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
                              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
                              "openThisPanel": "Open this panel.",
                              "pageUpPageDown": "page up and page down keys",
                              "questionMark": "Question mark",
                              "returnFocusToInput": "Return to the date input field.",
                              "selectFocusedDate": "Select the date in focus.",
                              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
                              "upArrowDownArrow": "up and down arrow keys",
                            }
          }
          readOnly={false}
          regular={false}
          renderCalendarDay={undefined}
          renderCalendarInfo={null}
          renderDayContents={null}
          renderMonth={null}
          reopenPickerOnClearDates={false}
          required={false}
          screenReaderInputMessage=""
          showClearDates={true}
          showDefaultInputIcon={false}
          small={false}
          startDate={"1970-01-01T00:00:00.000Z"}
          startDateId="startDate"
          startDatePlaceholderText="Start Date"
          transitionDuration={undefined}
          verticalHeight={null}
          verticalSpacing={22}
          weekDayFormat="dd"
          withFullScreenPortal={false}
          withPortal={false}
/>,
        <div
          className="logsListFilters__options"
>
          <select
                    className="logsListFilters__select"
                    onChange={[Function]}
                    value="newest"
          >
                    <option
                              value="newest"
                    >
                              new
                    </option>
                    <option
                              value="oldest"
                    >
                              old
                    </option>
          </select>
          <select
                    className="logsListFilters__select"
                    onChange={[Function]}
                    value={7}
          >
                    <option
                              value={1}
                    >
                              1
                    </option>
                    <option
                              value={7}
                    >
                              7
                    </option>
                    <option
                              value={14}
                    >
                              14
                    </option>
                    <option
                              value={31}
                    >
                              31
                    </option>
          </select>
</div>,
        <div
          className="logsListFilters__pages flex-right-end"
>
          <button
                    className="btn btn--page btn--disabled"
                    disabled={true}
                    onClick={[Function]}
          >
                    Prev
          </button>
          <button
                    className="btn btn--page btn--active"
                    disabled={true}
                    onClick={[Function]}
          >
                    1
          </button>
          <button
                    className="btn btn--page btn--disabled"
                    disabled={true}
                    onClick={[Function]}
          >
                    Next
          </button>
</div>,
      ],
      "className": "logsListFilters",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "anchorDirection": "left",
          "block": false,
          "calendarInfoPosition": "bottom",
          "customArrowIcon": null,
          "customCloseIcon": null,
          "customInputIcon": null,
          "daySize": 39,
          "disabled": false,
          "displayFormat": [Function],
          "enableOutsideDays": false,
          "endDate": "1970-01-01T00:00:01.000Z",
          "endDateId": "endDate",
          "endDatePlaceholderText": "End Date",
          "firstDayOfWeek": null,
          "focusedInput": null,
          "hideKeyboardShortcutsPanel": false,
          "horizontalMargin": 0,
          "initialVisibleMonth": null,
          "inputIconPosition": "before",
          "isDayBlocked": [Function],
          "isDayHighlighted": [Function],
          "isOutsideRange": [Function],
          "isRTL": false,
          "keepFocusOnInput": false,
          "keepOpenOnDateSelect": false,
          "minimumNights": 1,
          "monthFormat": "MMMM YYYY",
          "navNext": null,
          "navPrev": null,
          "noBorder": false,
          "numberOfMonths": 1,
          "onClose": [Function],
          "onDatesChange": [Function],
          "onFocusChange": [Function],
          "onNextMonthClick": [Function],
          "onPrevMonthClick": [Function],
          "openDirection": "down",
          "orientation": "horizontal",
          "phrases": Object {
            "calendarLabel": "Calendar",
            "chooseAvailableEndDate": [Function],
            "chooseAvailableStartDate": [Function],
            "clearDates": "Clear Dates",
            "closeDatePicker": "Close",
            "dateIsSelected": [Function],
            "dateIsUnavailable": [Function],
            "enterKey": "Enter key",
            "escape": "Escape key",
            "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
            "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
            "homeEnd": "Home and end keys",
            "jumpToNextMonth": "Move forward to switch to the next month.",
            "jumpToPrevMonth": "Move backward to switch to the previous month.",
            "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
            "keyboardShortcuts": "Keyboard Shortcuts",
            "leftArrowRightArrow": "Right and left arrow keys",
            "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
            "moveFocusByOneMonth": "Switch months.",
            "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
            "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
            "openThisPanel": "Open this panel.",
            "pageUpPageDown": "page up and page down keys",
            "questionMark": "Question mark",
            "returnFocusToInput": "Return to the date input field.",
            "selectFocusedDate": "Select the date in focus.",
            "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
            "upArrowDownArrow": "up and down arrow keys",
          },
          "readOnly": false,
          "regular": false,
          "renderCalendarDay": undefined,
          "renderCalendarInfo": null,
          "renderDayContents": null,
          "renderMonth": null,
          "reopenPickerOnClearDates": false,
          "required": false,
          "screenReaderInputMessage": "",
          "showClearDates": true,
          "showDefaultInputIcon": false,
          "small": false,
          "startDate": "1970-01-01T00:00:00.000Z",
          "startDateId": "startDate",
          "startDatePlaceholderText": "Start Date",
          "transitionDuration": undefined,
          "verticalHeight": null,
          "verticalSpacing": 22,
          "weekDayFormat": "dd",
          "withFullScreenPortal": false,
          "withPortal": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <select
              className="logsListFilters__select"
              onChange={[Function]}
              value="newest"
>
              <option
                            value="newest"
              >
                            new
              </option>
              <option
                            value="oldest"
              >
                            old
              </option>
</select>,
            <select
              className="logsListFilters__select"
              onChange={[Function]}
              value={7}
>
              <option
                            value={1}
              >
                            1
              </option>
              <option
                            value={7}
              >
                            7
              </option>
              <option
                            value={14}
              >
                            14
              </option>
              <option
                            value={31}
              >
                            31
              </option>
</select>,
          ],
          "className": "logsListFilters__options",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="newest"
>
                  new
</option>,
                <option
                  value="oldest"
>
                  old
</option>,
              ],
              "className": "logsListFilters__select",
              "onChange": [Function],
              "value": "newest",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "new",
                  "value": "newest",
                },
                "ref": null,
                "rendered": "new",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "old",
                  "value": "oldest",
                },
                "ref": null,
                "rendered": "old",
                "type": "option",
              },
            ],
            "type": "select",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value={1}
>
                  1
</option>,
                <option
                  value={7}
>
                  7
</option>,
                <option
                  value={14}
>
                  14
</option>,
                <option
                  value={31}
>
                  31
</option>,
              ],
              "className": "logsListFilters__select",
              "onChange": [Function],
              "value": 7,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "1",
                  "value": 1,
                },
                "ref": null,
                "rendered": "1",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "7",
                  "value": 7,
                },
                "ref": null,
                "rendered": "7",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "14",
                  "value": 14,
                },
                "ref": null,
                "rendered": "14",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "31",
                  "value": 31,
                },
                "ref": null,
                "rendered": "31",
                "type": "option",
              },
            ],
            "type": "select",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="btn btn--page btn--disabled"
              disabled={true}
              onClick={[Function]}
>
              Prev
</button>,
            Array [
              <button
                className="btn btn--page btn--active"
                disabled={true}
                onClick={[Function]}
>
                1
</button>,
            ],
            <button
              className="btn btn--page btn--disabled"
              disabled={true}
              onClick={[Function]}
>
              Next
</button>,
          ],
          "className": "logsListFilters__pages flex-right-end",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Prev",
              "className": "btn btn--page btn--disabled",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Prev",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": 1,
              "className": "btn btn--page btn--active",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": 1,
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next",
              "className": "btn btn--page btn--disabled",
              "disabled": true,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Next",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withStyles(DateRangePicker)
            anchorDirection="left"
            block={false}
            calendarInfoPosition="bottom"
            customArrowIcon={null}
            customCloseIcon={null}
            customInputIcon={null}
            daySize={39}
            disabled={false}
            displayFormat={[Function]}
            enableOutsideDays={false}
            endDate={"1970-01-01T00:00:01.000Z"}
            endDateId="endDate"
            endDatePlaceholderText="End Date"
            firstDayOfWeek={null}
            focusedInput={null}
            hideKeyboardShortcutsPanel={false}
            horizontalMargin={0}
            initialVisibleMonth={null}
            inputIconPosition="before"
            isDayBlocked={[Function]}
            isDayHighlighted={[Function]}
            isOutsideRange={[Function]}
            isRTL={false}
            keepFocusOnInput={false}
            keepOpenOnDateSelect={false}
            minimumNights={1}
            monthFormat="MMMM YYYY"
            navNext={null}
            navPrev={null}
            noBorder={false}
            numberOfMonths={1}
            onClose={[Function]}
            onDatesChange={[Function]}
            onFocusChange={[Function]}
            onNextMonthClick={[Function]}
            onPrevMonthClick={[Function]}
            openDirection="down"
            orientation="horizontal"
            phrases={
                        Object {
                                    "calendarLabel": "Calendar",
                                    "chooseAvailableEndDate": [Function],
                                    "chooseAvailableStartDate": [Function],
                                    "clearDates": "Clear Dates",
                                    "closeDatePicker": "Close",
                                    "dateIsSelected": [Function],
                                    "dateIsUnavailable": [Function],
                                    "enterKey": "Enter key",
                                    "escape": "Escape key",
                                    "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
                                    "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
                                    "homeEnd": "Home and end keys",
                                    "jumpToNextMonth": "Move forward to switch to the next month.",
                                    "jumpToPrevMonth": "Move backward to switch to the previous month.",
                                    "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
                          select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
                                    "keyboardShortcuts": "Keyboard Shortcuts",
                                    "leftArrowRightArrow": "Right and left arrow keys",
                                    "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
                                    "moveFocusByOneMonth": "Switch months.",
                                    "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
                                    "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
                                    "openThisPanel": "Open this panel.",
                                    "pageUpPageDown": "page up and page down keys",
                                    "questionMark": "Question mark",
                                    "returnFocusToInput": "Return to the date input field.",
                                    "selectFocusedDate": "Select the date in focus.",
                                    "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
                                    "upArrowDownArrow": "up and down arrow keys",
                                  }
            }
            readOnly={false}
            regular={false}
            renderCalendarDay={undefined}
            renderCalendarInfo={null}
            renderDayContents={null}
            renderMonth={null}
            reopenPickerOnClearDates={false}
            required={false}
            screenReaderInputMessage=""
            showClearDates={true}
            showDefaultInputIcon={false}
            small={false}
            startDate={"1970-01-01T00:00:00.000Z"}
            startDateId="startDate"
            startDatePlaceholderText="Start Date"
            transitionDuration={undefined}
            verticalHeight={null}
            verticalSpacing={22}
            weekDayFormat="dd"
            withFullScreenPortal={false}
            withPortal={false}
/>,
          <div
            className="logsListFilters__options"
>
            <select
                        className="logsListFilters__select"
                        onChange={[Function]}
                        value="newest"
            >
                        <option
                                    value="newest"
                        >
                                    new
                        </option>
                        <option
                                    value="oldest"
                        >
                                    old
                        </option>
            </select>
            <select
                        className="logsListFilters__select"
                        onChange={[Function]}
                        value={7}
            >
                        <option
                                    value={1}
                        >
                                    1
                        </option>
                        <option
                                    value={7}
                        >
                                    7
                        </option>
                        <option
                                    value={14}
                        >
                                    14
                        </option>
                        <option
                                    value={31}
                        >
                                    31
                        </option>
            </select>
</div>,
          <div
            className="logsListFilters__pages flex-right-end"
>
            <button
                        className="btn btn--page btn--disabled"
                        disabled={true}
                        onClick={[Function]}
            >
                        Prev
            </button>
            <button
                        className="btn btn--page btn--active"
                        disabled={true}
                        onClick={[Function]}
            >
                        1
            </button>
            <button
                        className="btn btn--page btn--disabled"
                        disabled={true}
                        onClick={[Function]}
            >
                        Next
            </button>
</div>,
        ],
        "className": "logsListFilters",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "anchorDirection": "left",
            "block": false,
            "calendarInfoPosition": "bottom",
            "customArrowIcon": null,
            "customCloseIcon": null,
            "customInputIcon": null,
            "daySize": 39,
            "disabled": false,
            "displayFormat": [Function],
            "enableOutsideDays": false,
            "endDate": "1970-01-01T00:00:01.000Z",
            "endDateId": "endDate",
            "endDatePlaceholderText": "End Date",
            "firstDayOfWeek": null,
            "focusedInput": null,
            "hideKeyboardShortcutsPanel": false,
            "horizontalMargin": 0,
            "initialVisibleMonth": null,
            "inputIconPosition": "before",
            "isDayBlocked": [Function],
            "isDayHighlighted": [Function],
            "isOutsideRange": [Function],
            "isRTL": false,
            "keepFocusOnInput": false,
            "keepOpenOnDateSelect": false,
            "minimumNights": 1,
            "monthFormat": "MMMM YYYY",
            "navNext": null,
            "navPrev": null,
            "noBorder": false,
            "numberOfMonths": 1,
            "onClose": [Function],
            "onDatesChange": [Function],
            "onFocusChange": [Function],
            "onNextMonthClick": [Function],
            "onPrevMonthClick": [Function],
            "openDirection": "down",
            "orientation": "horizontal",
            "phrases": Object {
              "calendarLabel": "Calendar",
              "chooseAvailableEndDate": [Function],
              "chooseAvailableStartDate": [Function],
              "clearDates": "Clear Dates",
              "closeDatePicker": "Close",
              "dateIsSelected": [Function],
              "dateIsUnavailable": [Function],
              "enterKey": "Enter key",
              "escape": "Escape key",
              "focusStartDate": "Interact with the calendar and add the check-in date for your trip.",
              "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
              "homeEnd": "Home and end keys",
              "jumpToNextMonth": "Move forward to switch to the next month.",
              "jumpToPrevMonth": "Move backward to switch to the previous month.",
              "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
              "keyboardShortcuts": "Keyboard Shortcuts",
              "leftArrowRightArrow": "Right and left arrow keys",
              "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
              "moveFocusByOneMonth": "Switch months.",
              "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
              "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
              "openThisPanel": "Open this panel.",
              "pageUpPageDown": "page up and page down keys",
              "questionMark": "Question mark",
              "returnFocusToInput": "Return to the date input field.",
              "selectFocusedDate": "Select the date in focus.",
              "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
              "upArrowDownArrow": "up and down arrow keys",
            },
            "readOnly": false,
            "regular": false,
            "renderCalendarDay": undefined,
            "renderCalendarInfo": null,
            "renderDayContents": null,
            "renderMonth": null,
            "reopenPickerOnClearDates": false,
            "required": false,
            "screenReaderInputMessage": "",
            "showClearDates": true,
            "showDefaultInputIcon": false,
            "small": false,
            "startDate": "1970-01-01T00:00:00.000Z",
            "startDateId": "startDate",
            "startDatePlaceholderText": "Start Date",
            "transitionDuration": undefined,
            "verticalHeight": null,
            "verticalSpacing": 22,
            "weekDayFormat": "dd",
            "withFullScreenPortal": false,
            "withPortal": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <select
                className="logsListFilters__select"
                onChange={[Function]}
                value="newest"
>
                <option
                                value="newest"
                >
                                new
                </option>
                <option
                                value="oldest"
                >
                                old
                </option>
</select>,
              <select
                className="logsListFilters__select"
                onChange={[Function]}
                value={7}
>
                <option
                                value={1}
                >
                                1
                </option>
                <option
                                value={7}
                >
                                7
                </option>
                <option
                                value={14}
                >
                                14
                </option>
                <option
                                value={31}
                >
                                31
                </option>
</select>,
            ],
            "className": "logsListFilters__options",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value="newest"
>
                    new
</option>,
                  <option
                    value="oldest"
>
                    old
</option>,
                ],
                "className": "logsListFilters__select",
                "onChange": [Function],
                "value": "newest",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "new",
                    "value": "newest",
                  },
                  "ref": null,
                  "rendered": "new",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "old",
                    "value": "oldest",
                  },
                  "ref": null,
                  "rendered": "old",
                  "type": "option",
                },
              ],
              "type": "select",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <option
                    value={1}
>
                    1
</option>,
                  <option
                    value={7}
>
                    7
</option>,
                  <option
                    value={14}
>
                    14
</option>,
                  <option
                    value={31}
>
                    31
</option>,
                ],
                "className": "logsListFilters__select",
                "onChange": [Function],
                "value": 7,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "1",
                    "value": 1,
                  },
                  "ref": null,
                  "rendered": "1",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "7",
                    "value": 7,
                  },
                  "ref": null,
                  "rendered": "7",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "14",
                    "value": 14,
                  },
                  "ref": null,
                  "rendered": "14",
                  "type": "option",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "31",
                    "value": 31,
                  },
                  "ref": null,
                  "rendered": "31",
                  "type": "option",
                },
              ],
              "type": "select",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="btn btn--page btn--disabled"
                disabled={true}
                onClick={[Function]}
>
                Prev
</button>,
              Array [
                <button
                  className="btn btn--page btn--active"
                  disabled={true}
                  onClick={[Function]}
>
                  1
</button>,
              ],
              <button
                className="btn btn--page btn--disabled"
                disabled={true}
                onClick={[Function]}
>
                Next
</button>,
            ],
            "className": "logsListFilters__pages flex-right-end",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Prev",
                "className": "btn btn--page btn--disabled",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Prev",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": 1,
                "className": "btn btn--page btn--active",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": 1,
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Next",
                "className": "btn btn--page btn--disabled",
                "disabled": true,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Next",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
